Add a list of users not attending the gathering

hardfire opened the pull request:
I took at stab at generating the list of users who are not attending the gathering. 
The corresponding changes in patchbay-gathering is in [my fork](https://github.com/hardfire/patchbay-gatherings) . 

It works in theory, but some things could be done better (especially the naming convention of "unattending"). A quick [search shows](https://english.stackexchange.com/questions/261805/the-opposite-of-attend) there isn't really an opposite of attending.

hardfire committed:
add an array of people not attending the gathering

hardfire commented:
@mixmix can't add you as a reviewer/assignee so hopefully this comment will notify you!

mixmix commented:
Cool I'll check this out when I'm at my laptop!

On Sun, 7 Oct 2018, 17:43 अभिनाश (Avinash), <notifications@github.com>
wrote:

> @mixmix <https://github.com/mixmix> can't add you as a reviewer/assignee
> so hopefully this comment will notify you!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ssbc/scuttle-gathering/pull/2#issuecomment-427625510>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACitni9FYSaiIW-ZqcJwJO5_LGDEBeFjks5uiYZsgaJpZM4XLZEJ>
> .
>

mixmix commented:
the logic looks good to me.
I think for tests I'd like to see a case where a person is attending, then marks themself as not attending and the state looks correct.
This logic is kinda simple, but I think it could be a good exercise in understanding how to write tests and it will make sure refactoring in the future is easy. I think I would have make the accumulator an object instead of an Array, but I don't think it matters really

As for naming, I agree it's ugly. I think you should change it to `attendees` and `notAttendees`

---

I'll add you as an owner of this repo so you don't have to work on a branch and we can hack on this PR together more easily!
I'll do likewise with with patchbay-gatherings (if I have access), because that's I think where you'll b going next

hardfire commented:
Creating a new pull request with a branch in this repository. 
- switched naming to notAttendees
- the test does as mentioned, where the person is attending and then marks themself as not attending. It was there before, i just added the check for the notAttending cases. 
and closing this :D
